crates/core/tests/integration.rs

```diff
@@ -0,0 +1,71 @@
+use std::{fs::File, path::Path, sync::Arc};
+
+use anyhow::Result;
+use flate2::read::GzDecoder;
+use pretty_assertions::assert_eq;
+use rustic_core::{
+    repofile::SnapshotFile, BackupOptions, ConfigOptions, InMemoryBackend, KeyOptions,
+    NoProgressBars, OpenStatus, PathList, Repository, RepositoryBackends, RepositoryOptions,
+};
+use tar::Archive;
+use tempfile::{tempdir, TempDir};
+
+fn set_up_repo() -> Result<Repository<NoProgressBars, OpenStatus>> {
+    let be = InMemoryBackend::new();
+    let be = RepositoryBackends::new(Arc::new(be), None);
+    let options = RepositoryOptions::default().password("test");
+    let repo = Repository::new(&options, be)?;
+    let key_opts = KeyOptions::default();
+    let config_opts = &ConfigOptions::default();
+    let repo = repo.init(&key_opts, config_opts)?;
+    Ok(repo)
+}
+
+fn set_up_tar_fixture(path: impl AsRef<Path>) -> Result<TempDir> {
+    let dir = tempdir()?;
+    let path = Path::new("tests/testdata").join(path);
+    let tar_gz = File::open(path)?;
+    let tar = GzDecoder::new(tar_gz);
+    let mut archive = Archive::new(tar);
+    archive.unpack(&dir)?;
+    Ok(dir)
+}
```

Comment:
https://crates.io/crates/rstest fixtures are also nice for this, it works via DI